### DISCUSSION TITLE
Blasting through Amazon S3 Get Requests & Put, Copy, Post or List Requests  (Hubs Clouds)

### DISCUSSION THREAD

**Discussion by Aaron-E-Walsh** [mozilla/hubs#3318] [Hubs Cloud Questions] [answered]
Greetings, fellow Hub Clouds admins and developers.

We have deployed Hubs Clouds on a trial Amazon AWS account, to test Hubs and also help us better understand the pricing as Hubs is utilized by small groups.

All is going well, but I have been very surprised to see (and be notified by Amazon) that we immediately blasted through our free trial monthly limits of the following services within just 2 full hours of using Hubs:

Amazon Simple Storage | 20,000 Get Requests of Amazon S3 (EXCEEDED IN 2 HOURs OF USING HUBS)
Amazon Simple Storage | 2,000 Put, Copy, Post or List Requests of Amazon S3 (90% REACHED IN 2 HOURS)

This has happened twice since we installed Hubs Cloud two month ago (these monthly limits were immediately exceeded in the month of October, and then again in November).

We used one of the Hubs meeting spaces (a large space, the one with the **fishtank** in the scene and images on the walls), and approximately 10 of us met in that space for approximately 2 hours last month (Oct) and this month.  Both times we receive a notice of exceeding the above monthly Amazon free trial limits after the 2 hour meetings.

Is this typical, or something to be concerned about?

With thanks for your insights as we continue to enjoy and test Hubs.

Aaron

**Comment by gfodor** [comment, marked as answer]
This happens because the backend server continually checks the S3 bucket for changes to the client code. This was already optimized somewhat but could be reduced much further for Hubs Cloud users since they are not having updated client code frequently. (The Mozilla team deployed code frequently enough that we wanted it to update right away, which required this polling.)

**Comment by Aaron-E-Walsh** [comment]
Ah, I see.  Thank you very much for your reply, Greg. This now makes sense.
I was concerned that perhaps our installation wasn't functioning properly,
but am glad to hear that this activity is intentional and accounted for.

Yes, as you have suggested it would be quite helpful to have an option
(such as a setting in the administration area) to set the timing of these
checks in order to conserve server and network resources when they are not
required at such a high frequency.

With thanks again for your reply,
Aaron

**Reply by robinkwilson** [reply]
If you'd like to see this feature in the future! Create an issue/feature request in https://github.com/mozilla/hubs-cloud please!

**Reply by Aaron-E-Walsh** [reply]
Will do!

Aaron
